### PR TITLE
Saved filters CRUD API

### PR DESCRIPTION
**Pull request by lukaa12**

### CODE REVIEW

File: kentik_api_library/kentik_api/api_resources/saved_filters_api.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+class SavedFiltersAPI:
+    """Exposes Kentik API operations related to saved filters. """
+
+    def __init__(self, api_connector: APIConnectorProtocol) -> None:
+        self._api_connector = api_connector
+
+    def get_all(self) -> List[SavedFilter]:
+        api_call = saved_filters.get_saved_filters()
+        response = self._api_connector.send(api_call)
+        print(response)
```
Comment (mateuszmidor) [inline kentik_api_library/kentik_api/api_resources/saved_filters_api.py:20]: probably should be removed

File: kentik_api_library/kentik_api/api_resources/saved_filters_api.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        return saved_filters_payload.GetAllResponse.from_json(response.text).to_saved_filters()
+
+    def get(self, saved_filter_id: int) -> SavedFilter:
+        api_call = saved_filters.get_saved_filter_info(saved_filter_id)
+        response = self._api_connector.send(api_call)
+        print(response)
```
Comment (mateuszmidor) [inline kentik_api_library/kentik_api/api_resources/saved_filters_api.py:26]: probably should be removed

File: kentik_api_library/kentik_api/api_resources/saved_filters_api.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        return saved_filters_payload.GetResponse.from_json(response.text).to_saved_filter()
+
+    def create(self, saved_filter: SavedFilter) -> SavedFilter:
+        SavedFiltersAPI.check_fields(saved_filter)
+
+        api_call = saved_filters.create_saved_filter()
+        payload = as_dict(saved_filters_payload.CreateRequest(saved_filter))
+        response = self._api_connector.send(api_call, payload)
+        print(response)
```
Comment (mateuszmidor) [inline kentik_api_library/kentik_api/api_resources/saved_filters_api.py:35]: probably should be removed

File: kentik_api_library/kentik_api/api_resources/saved_filters_api.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        return saved_filters_payload.CreateResponse.from_json(response.text).to_saved_filter()
+
+    def update(self, saved_filter: SavedFilter) -> SavedFilter:
+        assert saved_filter.id is not None, "SavedFilter ID has to be provided"
+        SavedFiltersAPI.check_fields(saved_filter)
+
+        api_call = saved_filters.update_saved_filter(saved_filter.id)
+        payload = as_dict(saved_filters_payload.UpdateRequest(saved_filter))
+        response = self._api_connector.send(api_call, payload)
+        print(response)
```
Comment (mateuszmidor) [inline kentik_api_library/kentik_api/api_resources/saved_filters_api.py:45]: probably should be removed

File: kentik_api_library/kentik_api/api_resources/saved_filters_api.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        return saved_filters_payload.UpdateResponse.from_json(response.text).to_saved_filter()
+
+    def delete(self, saved_filter_id: int) -> bool:
+        api_call = saved_filters.delete_saved_filter(saved_filter_id)
+        response = self._api_connector.send(api_call)
+        print(response)
```
Comment (mateuszmidor) [inline kentik_api_library/kentik_api/api_resources/saved_filters_api.py:51]: probably should be removed

**Review by mateuszmidor — COMMENTED** [review]
Please remove print statements, other than that the code looks good!

File: kentik_api_library/kentik_api/api_resources/saved_filters_api.py (hunk 1 above)
Comment (lukaa12) [reply kentik_api_library/kentik_api/api_resources/saved_filters_api.py]: Yes, I will remove it.

File: kentik_api_library/kentik_api/api_resources/saved_filters_api.py (hunk 2 above)
Comment (lukaa12) [reply kentik_api_library/kentik_api/api_resources/saved_filters_api.py]: Yes, I agree

File: kentik_api_library/kentik_api/api_resources/saved_filters_api.py (hunk 3 above)
Comment (lukaa12) [reply kentik_api_library/kentik_api/api_resources/saved_filters_api.py]: ack

File: kentik_api_library/kentik_api/api_resources/saved_filters_api.py (hunk 4 above)
Comment (lukaa12) [reply kentik_api_library/kentik_api/api_resources/saved_filters_api.py]: ack

File: kentik_api_library/kentik_api/api_resources/saved_filters_api.py (hunk 5 above)
Comment (lukaa12) [reply kentik_api_library/kentik_api/api_resources/saved_filters_api.py]: ack

**Review by mateuszmidor — APPROVED** [review]

**Review by adrian-szczepanski — APPROVED** [review]

**Review by TedatKentik — APPROVED** [review]

**Pull request merged**